### ISSUE TITLE
Support short DOI names

### ISSUE THREAD

**Issue by bwiernik** [Impactstory/oadoi#3]
The search page doesn't appear to support short DOI names (http://shortdoi.org)-- it shows the "can't type, must paste" error when I try to past a short DOI.

e.g., 10/895

**Comment by jasonpriem** [comment]
Thanks! Alas supporting shortDOI is not this simple. We are using the DOI prefixes to help us determine which DOIs are DataCite ones rather than CrossRef ones, which is in turn necessary to know which APIs to call to resolve them. So, when we accept a shortDOI, we'll need to also do an extra lookup to get the canonical version with its prefix info.

This can, of course, be done. It's not that hard. However, it's extra work to implement, extra code surface area, and extra latency on each request. So we'd like to wait on this one for now. Once we've heard a few more people +1 shortDOI support we'll bump it up in the priority queue. Will leave this issue open to see if it gets more traction.

In the meantime we will supply a more useful error message at least, the current one sucks for this use case. 

Thanks again for flagging this up, and for the PR.

**Comment by bwiernik** [comment]
One solution to support shortDOI would be to query the DOI API to get then long DOI, then pass this to the rest of the oaDOI functions.

So, if a "10/" string is detected `inputVal.match(/10\/[^\s]*[^\s\.,]/)`, extract the shortDOI from the input, then pass it to a function like this:
```
var url = 'https://doi.org/api/handles/' + encodeURIComponent(shortDOI); 
var req = new XMLHttpRequest();
req.open('GET', url, true);
req.responseType = 'json';

req.onreadystatechange = function() { 
  if (req.readyState == 4) {
    if (req.status == 200) {
      if (req.response.responseCode == 1) {
        var longDOI = req.response.values["1"].data.value;
        return longDOI;
      } else if (req.status == 404) {
        //invalid DOI response
      } else {
        //no response
      }
    }
  }
};

req.send(null);
```

**Comment by jasonpriem** [comment]
@bwiernik Good idea! However, the oaDOI page in question is no longer up.

**Comment by jasonpriem** [comment]
Ah, understood. Yes, good point...in fact pretty much everything Unpaywall does (Unpaywall is the new name for oaDOI, which we need to change on this repository, oops...) fails on shortDOIs. We have found they are just not used enough in the wild, particularly for our use cases, to justify support.

I like your solution for low-volume cases, but unfortunately I don't think it's practical for the scale we're supporting right now....we have a completely open API (no API keys, no IP-based throttling) and we get millions of requests daily. So we need to avoid putting anything that requires something as (relatively) slow as a call out to Crossref in front of an API return. We used to do that kind of hot-calling and it scaled very poorly. If someone wanted to throw a million shortDOIs at us in a few hours, now we have to make a million calls out to Crossref, and not only is that going to challenge our infrastructure, it's not great for them either.

So, in order to integrate shortDOIs, the only practical source I can think of would be to bring every one of them into our database and index them all beforehand. I think this is what we'll do if there's ever enough demand to support them.
